components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java

```diff
@@ -276,6 +291,19 @@ void checkConnection() throws IOException {
                 mllpSocketWriter = new MllpSocketWriter(socket, false);
             }
         }
+        return;
+    }
+
+    @ManagedOperation(description = "Check client connection")
+    public boolean managedCheckConnection() {
+        boolean isValid = true;
+        try {
+            checkConnection();
+        } catch (IOException ioEx) {
+            isValid = false;
+            log.error("JMX check connection: {}", ioEx);
```

Comment:
Are you sure that you want to log an error there ? At first read, it's like this is just nominal case.